models/AbstractModel.php

```diff
@@ -53,6 +53,30 @@ protected function getDataMapper():AbstractDataMapper {
             return $this->relatedMapper;
         }
 
+        /**
+         * {@inheritDoc}
+         */
+        public function getList(int $limit, int $offset):array {
+            $connection = $this->connectToDB();
+
+            $query      = (new SelectQueryBuilder($this->getTableName()))
+                            ->whereAnd("`" . $this->getDomainName() . "_is_deleted` = 0")
+                            ->limit($limit, $offset)
+                            ->build();
+
+            $result    = $connection->query($query->getQueryString());
+            $itemsList = $result->fetch_all(MYSQLI_ASSOC);
+
+            return $itemsList;
+        }
```

Comment:
This method looked almost the same in all models which implemented IController. The only things which were different were the List name and the query to DB, so the method was generalised with a promise to implement an abstract function in child classes.